### ISSUE TITLE
[BUG] 403 error while using Pytube on a server but it works fine locally.

### ISSUE THREAD

**Issue by matildasmets** [pytube/pytube#1850]
**Before creating an issue**

Please confirm that you are on the latest version of pytube by installing from the source.
You can do this by running `python -m pip install git+https://github.com/pytube/pytube`.
Sometimes, the pypi library repository is not up to date, and your issue may have been fixed already!

I'm using the latest.

**Describe the bug**

So, I have a Discord bot that uses the URL to play the audio of the youtube video. But since recently (it did work before, but not anymore) the bot keeps returning a 403 error while trying to play the audio: 

```
[https @ 0x559e77891540] HTTP error 403 Forbidden
31.12 20:05:15 [Bot] https://rr2---sn-4g5edn6y.googlevideo.com/videoplayback?expire=1704074713&ei=ecmRZYWVGOKHjMwP9pCkwA8&ip=135.125.16.255&id=o-APfEzE1H-bjWwFEEpvK-Z1w0MTfZZrmPuIHzARPx5ZWI&itag=139&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=mq&mm=31%2C26&mn=sn-4g5edn6y%2Csn-25ge7nzk&ms=au%2Conr&mv=m&mvi=2&pl=20&initcwndbps=176250&vprv=1&mime=audio%2Fmp4&gir=yes&clen=568807&dur=93.158&lmt=1692364350722555&mt=1704052612&fvip=4&keepalive=yes&fexp=24007246&c=ANDROID_EMBEDDED_PLAYER&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAJrOpUaze0wOPTH3psH2ENP62UVlFNJbrzVxkS_VDUbEAiBx-EmSXRhyK-vaiRI9hDD0Q3OZGseRlUtcQCsAn_qpMA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AAO5W4owRgIhAPTDTvfnrpfrTg2M9CE7do5C3dHzM3_ena6nyI-ccOHgAiEA75W-c6fYvgp6uU1BBT9mKOdnG5lrktfv0zNHctG3HFQ%3D: Server returned 403 Forbidden (access denied)
```

This is my code (partially):
```py
try:
              headers = {
                  'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3',
              }

              response = requests.get(url, headers=headers)

              yt = pytube.YouTube(response.url)
              video = yt.streams.filter(only_audio=True).first()
              audio_stream_url = video.url
```

i tried adding headers but those didn't make any differences. The audio_stream_url is causing the 403 issue. But locally, the audio_stream_url doesn't cause any issues. Only when I try executing this code on my server.

**To Reproduce**
Please provide the following information:
- The video or playlist url that is causing the error.
- The code where the problem is occurring.

Video doesn't matter. It does it for all videos (on my server). I'm using Pebblehost.

**Expected behavior**

The expected behavior is that it uses the URL accordingly and plays the audio.

**Output**
If Pytube raises an exception, please provide the full traceback for the exception.

```
[https @ 0x559e77891540] HTTP error 403 Forbidden
31.12 20:05:15 [Bot] https://rr2---sn-4g5edn6y.googlevideo.com/videoplayback?expire=1704074713&ei=ecmRZYWVGOKHjMwP9pCkwA8&ip=135.125.16.255&id=o-APfEzE1H-bjWwFEEpvK-Z1w0MTfZZrmPuIHzARPx5ZWI&itag=139&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=mq&mm=31%2C26&mn=sn-4g5edn6y%2Csn-25ge7nzk&ms=au%2Conr&mv=m&mvi=2&pl=20&initcwndbps=176250&vprv=1&mime=audio%2Fmp4&gir=yes&clen=568807&dur=93.158&lmt=1692364350722555&mt=1704052612&fvip=4&keepalive=yes&fexp=24007246&c=ANDROID_EMBEDDED_PLAYER&txp=5532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAJrOpUaze0wOPTH3psH2ENP62UVlFNJbrzVxkS_VDUbEAiBx-EmSXRhyK-vaiRI9hDD0Q3OZGseRlUtcQCsAn_qpMA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AAO5W4owRgIhAPTDTvfnrpfrTg2M9CE7do5C3dHzM3_ena6nyI-ccOHgAiEA75W-c6fYvgp6uU1BBT9mKOdnG5lrktfv0zNHctG3HFQ%3D: Server returned 403 Forbidden (access denied)
```

**System information**
Please provide the following information:
- Python version: 3.9
- Pytube version: 15.0.0

**Comment by bayhax** [comment]
same, did you fix it?

**Comment by matildasmets** [comment]
> same, did you fix it?

Nope, never found a way to fix it or else I would post a comment here about it. Anyway, I have a speculation why it might not work:

- Locally you only make a few requests to Youtube. This doesn't seem suspicious, it can be done.
- After uploading the code it might work fine for a while, but this is for my Discord bot so my bot reaches around 500 servers. 
- The requests for music go up (due to popularity), sending more data scraping requests towards Youtube. 
- They think it's suspicious and deny the requests.

Very annoying. I tried to add headers to make the request look unique, didn't work.

**Comment by bayhax** [comment]
> > same, did you fix it?
> 
> Nope, never found a way to fix it or else I would post a comment here about it. Anyway, I have a speculation why it might not work:
> 
> * Locally you only make a few requests to Youtube. This doesn't seem suspicious, it can be done.
> * After uploading the code it might work fine for a while, but this is for my Discord bot so my bot reaches around 500 servers.
> * The requests for music go up (due to popularity), sending more data scraping requests towards Youtube.
> * They think it's suspicious and deny the requests.
> 
> Very annoying. I tried to add headers to make the request look unique, didn't work.

Hi, After my one day testing, I found out that it's the ip problem, first of all, you have to keep your ip clean so that the download can work properly, you can ko rate to use vpn, ip polling mechanism.

**Comment by matildasmets** [comment]
> > > same, did you fix it?
> > 
> > 
> > Nope, never found a way to fix it or else I would post a comment here about it. Anyway, I have a speculation why it might not work:
> > 
> > * Locally you only make a few requests to Youtube. This doesn't seem suspicious, it can be done.
> > * After uploading the code it might work fine for a while, but this is for my Discord bot so my bot reaches around 500 servers.
> > * The requests for music go up (due to popularity), sending more data scraping requests towards Youtube.
> > * They think it's suspicious and deny the requests.
> > 
> > Very annoying. I tried to add headers to make the request look unique, didn't work.
> 
> Hi, After my one day testing, I found out that it's the ip problem, first of all, you have to keep your ip clean so that the download can work properly, you can ko rate to use vpn, ip polling mechanism.

I don't understand how to use VPNs/proxies in my code and where to find them. How did you do it? And, I want to fix this but I don't want to spend (lots of) money on proxies or vpn's since it's a free Discord bot. I don't get paid.

**Comment by bayhax** [comment]
> > > > same, did you fix it?
> > > 
> > > 
> > > Nope, never found a way to fix it or else I would post a comment here about it. Anyway, I have a speculation why it might not work:
> > > 
> > > * Locally you only make a few requests to Youtube. This doesn't seem suspicious, it can be done.
> > > * After uploading the code it might work fine for a while, but this is for my Discord bot so my bot reaches around 500 servers.
> > > * The requests for music go up (due to popularity), sending more data scraping requests towards Youtube.
> > > * They think it's suspicious and deny the requests.
> > > 
> > > Very annoying. I tried to add headers to make the request look unique, didn't work.
> > 
> > 
> > Hi, After my one day testing, I found out that it's the ip problem, first of all, you have to keep your ip clean so that the download can work properly, you can ko rate to use vpn, ip polling mechanism.
> 
> I don't understand how to use VPNs/proxies in my code and where to find them. How did you do it? And, I want to fix this but I don't want to spend (lots of) money on proxies or vpn's since it's a free Discord bot. I don't get paid.

you can use proxies like `yt = YouTube(youtube_url, proxies=proxies)`. But if you don't want to spend money. it's no way out.  you can try to find some free proxies like oculusproxies.com
